### ISSUE TITLE
speedup #usingMethods for class variables

### ISSUE THREAD

**Issue by MarcusDenker** [guillep/pharo-core#3914]
we can speedup #usingMethods for class variables: we do not need to search all methods, but just the hierarchy.

This change suggests, too, that we should model bindings coming from a Pool as it's own kind of LiteralVariable. This is future work.

[(SmalltalkImage classVariableNamed: #CompilerClass) usingMethods] bench

before: '17.324 per second'"
after:  "'63.049 per second'" "
